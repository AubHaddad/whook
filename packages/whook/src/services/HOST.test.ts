import { jest } from '@jest/globals';
import { internalIpV4 } from 'internal-ip';
import initHOST from './HOST.js';
import type { LogService } from 'common-services';
import type { ImporterService } from './importer.js';

describe('initHOST', () => {
  const log = jest.fn<LogService>();
  const importer = jest.fn<ImporterService<any>>();
  const internalIp = { internalIpV4: jest.fn<typeof internalIpV4>() };

  beforeEach(() => {
    log.mockReset();
    importer.mockReset();
    internalIp.internalIpV4.mockReset();
  });

  it('should use the env HOST first', async () => {
    importer.mockResolvedValueOnce(internalIp);

    const HOST = await initHOST({
      ENV: { HOST: '192.168.1.11' },
      log,
      importer,
    });

    expect(HOST).toMatchInlineSnapshot(`"192.168.1.11"`);
    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      requireCalls: importer.mock.calls,
      internalIpV4Calls: internalIp.internalIpV4.mock.calls,
    }).toMatchSnapshot();
  });

  it('should find a HOST by itself if no env HOST', async () => {
    importer.mockResolvedValueOnce(internalIp);
    internalIp.internalIpV4.mockResolvedValueOnce('192.168.1.10');

    const HOST = await initHOST({
      ENV: {},
      log,
      importer,
    });

    expect(HOST).toMatchInlineSnapshot(`"192.168.1.10"`);
    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      requireCalls: importer.mock.calls,
      internalIpV4Calls: internalIp.internalIpV4.mock.calls,
    }).toMatchSnapshot();
  });

  it('should fallback to localhost', async () => {
    importer.mockResolvedValueOnce(internalIp);
    internalIp.internalIpV4.mockResolvedValueOnce('');

    const HOST = await initHOST({
      ENV: {},
      log,
      importer,
    });

    expect(HOST).toMatchInlineSnapshot(`"localhost"`);
    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      requireCalls: importer.mock.calls,
      internalIpV4Calls: internalIp.internalIpV4.mock.calls,
    }).toMatchSnapshot();
  });
});
