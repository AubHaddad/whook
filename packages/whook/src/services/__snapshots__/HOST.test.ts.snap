// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initHOST should fallback to localhost 2`] = `
{
  "internalIpV4Calls": [
    [],
  ],
  "logCalls": [
    [
      "debug",
      "üè≠ - Initializing the HOST service.",
    ],
    [
      "warning",
      "üö´ - Could not detect any host. Fallbacking to "localhost".",
    ],
  ],
  "requireCalls": [
    [
      "internal-ip",
    ],
  ],
}
`;

exports[`initHOST should find a HOST by itself if no env HOST 2`] = `
{
  "internalIpV4Calls": [
    [],
  ],
  "logCalls": [
    [
      "debug",
      "üè≠ - Initializing the HOST service.",
    ],
    [
      "warning",
      "‚úî - Using detected host "192.168.1.10".",
    ],
  ],
  "requireCalls": [
    [
      "internal-ip",
    ],
  ],
}
`;

exports[`initHOST should use the env HOST first 2`] = `
{
  "internalIpV4Calls": [],
  "logCalls": [
    [
      "debug",
      "üè≠ - Initializing the HOST service.",
    ],
    [
      "warning",
      "‚ôªÔ∏è - Using ENV host "192.168.1.11"",
    ],
  ],
  "requireCalls": [],
}
`;
